### PR TITLE
fix: Update the S3Storage to work with both node version 8.x and 10.x

### PR DESCRIPTION
**Pull request by etienne95**
We modified the S3Storage file in order to read the readable object length both in node version 8.x and 10.x. We did that because when deployed an AWS handler that uploads a file in a bucket, the readable object in the 8.x node version doesn't have the readableLength property.

## What did you implement:

We updated the S3Storage to work with both node version 8.x and 10.x.

## How did you implement it:

We modified the S3Storage file in order to read the readable object length both in node version 8.x and 10.x. We did that because when deployed an AWS handler that uploads a file in a bucket, the readable object in the 8.x node version doesn't have the readableLength property.

## How can we verify it:

_Unit tests_

image

## Todos:

_**Note: Run `npm run test-ci` to run all validation checks on proposed changes**_

- [X] Write tests and confirm existing functionality is not broken.
       **Validate via `npm test`**
- [X] Write documentation
- [X] Ensure there are no lint errors.
       **Validate via `npm run lint-updated`**
       _Note: Some reported issues can be automatically fixed by running `npm run lint:fix`_
- [X] Ensure introduced changes match Prettier formatting.
       **Validate via `npm run prettier-check-updated`**
       _Note: All reported issues can be automatically fixed by running `npm run prettify-updated`_
- [X] Make sure code coverage hasn't dropped
- [ ] Provide verification config / commands / resources
- [X] Enable "Allow edits from maintainers" for this PR
- [X] Update the messages below

**_Is this ready for review?:_** YES
**_Is it a breaking change?:_** NO

### CODE REVIEW

File: aws/src/services/S3Storage.ts [hunk 1]
```diff
@@ -53,7 +53,7 @@ export class S3Storage implements CloudStorage {
       Bucket: opts.container,
       Key: opts.path,
       Body: streamBody,
-      ContentLength: streamBody.readableLength
+      ContentLength: streamBody.readableLength || (streamBody as any)._readableState.length
```
Comment (lgaspari) [inline aws/src/services/S3Storage.ts]: I don't like the idea of accessing private properties we may investigate further about this

File: aws/src/services/S3Storage.test.ts [hunk 2]
```diff
@@ -125,7 +125,7 @@ describe("aws storage when call write should", () => {
       Bucket: input.container,
       Key: input.path,
       Body: readableBody,
-      ContentLength: readableBody.readableLength
+      ContentLength: readableBody.readableLength || (readableBody as any)._readableState.length
```
Comment (wbreza) [inline aws/src/services/S3Storage.test.ts]: Let's see what other solutions we can find.  The `readableState` is not exposed as a public member somewhere?

**Review by wbreza — CHANGES_REQUESTED** [review]
Let's see what other solutions we can find vs accessing the private members.

**Comment by etienne95** [comment]
We will close this PR because we will use the node.js 10.x version.

**Pull request closed without merging**